rfd/0103-application-access-web-ui-auth-flow.md

```diff
@@ -22,53 +21,44 @@ the auth service. This application session has two values (a name and a bearer t
 as cookies when the user navigates to the application's URL.
 
 The flow to be able to set the needed cookies from the application session across different domains is
-as follows:
+as follows (using debug dumper app as example):
 
 ```mermaid
 sequenceDiagram
-    participant User
-    participant Proxy UI
-    participant Application
+    participant Browser
+    participant Proxy
+    participant Target App
     participant Auth Service
-    User->>Proxy UI: Launch Application
-    Proxy UI->>Auth Service: Create App Session /v1/webapi/sessions/app
-    Auth Service->>Proxy UI: Returns two cookie values (app session & subject)
-    Proxy UI->>Application: CORS preflight request to appurl.com/x-teleport-auth
-    Application->>Proxy UI: Matches Origin against public proxy addresses & allows CORS if a match
-    Proxy UI->>Application: POST appurl.com/x-teleport-auth with the cookie values in the header
-    Application->>Proxy UI: Sets the cookies on appurl.com if they are valid
-    Proxy UI->>User: User is logged in, redirect to appurl.com
-```
-
-### CORS
-
-As we need to enable cross-origin requests from the proxy URL to the application URL, we add middleware
-before `/x-teleport-auth` to check the `Origin` header (this cannot be changed via a `fetch` request).
-
-We check the origin against the public addresses of the proxy, and if one is a match, we allow a cross-origin
-request.
-
-The rules around the cross-origin request are strict and the absolute minimum needed:
+    alt Loading target app outside of web UI
+    Note left of Browser: User copy pastes target app URL into browser <br/>https://dumper.localhost:3080
+    Browser->>Proxy: Redirect to web launcher, preserving the requested path and query params <br/>In subsequent requests both the path and query gets preserved as path<br/>REDIRECT /web/launch/dumper.localhost?path=<requested-path>&query=<requested-query>
+    Note left of Browser: Redirected to app launcher
+    else Loading target app within the web UI (using the web app launcher)
+    Note left of Browser: At web UI, user clicks on `launch` button for the target app
+    Note left of Browser: Web UI routes to app launcher <br/>/web/launch/dumper.localhost/cluster-name/dumper.localhost<br/>route format: /web/launch/:fqdn/:clusterId?/:publicAddr?
+    end
+    Note right of Browser: App launcher navigates to target app <br/>https://dumper.localhost:3080/x-teleport-auth
+    Browser->>Target App: Navigating to target app start's the auth exchange <br/>GET /dumper.localhost/x-teleport-auth

```

Comment:
The target app doesn't know anything about `x-teleport-auth` though. Should this arrow be pointed at the proxy instead of the target app?

Reply:
sorry, in my head i saw them as "app handlers", i clarified like this:

<img width="178" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/89168c42-be5f-4395-962c-0b4a34019e07">